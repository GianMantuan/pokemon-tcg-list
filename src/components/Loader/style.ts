import styled from 'styled-components/native';

export const LoaderContainer = styled.View`
  flex: 1
  flex-direction: column
  align-self: center
  
`;

export const LoaderText = styled.Text`
  font-size: 16px;
  color: #121212
  font-weight: 300
`;
