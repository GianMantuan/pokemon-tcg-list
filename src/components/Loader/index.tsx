import React from 'react';
import {ActivityIndicator} from 'react-native';

import {LoaderContainer, LoaderText} from './style';

type Props = {
  text?: string;
};

export default function Loader({text}: Props): JSX.Element {
  return (
    <LoaderContainer>
      <ActivityIndicator size={36} />
      {text ? <LoaderText>{text}</LoaderText> : null}
    </LoaderContainer>
  );
}
