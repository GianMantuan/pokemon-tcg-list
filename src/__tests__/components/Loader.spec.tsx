import React from 'react'
import { Text } from 'react-native'
import Loader from '../../components/Loader'

import { render, cleanup } from '@testing-library/react-native'
import renderer from 'react-test-renderer'

describe('Loader Component', () => {
  afterEach(cleanup)

  it('should render Loader component', () => {
    expect(renderer.create(<Loader />)).toMatchSnapshot()
  })

  it('should not have text rendered as a child', () => {
    const loader = renderer.create(<Loader />)

    expect(loader.root.props).toStrictEqual({})
  })

  it('should have text rendered as a child', () => {
    const loader = renderer.create(<Loader text="loading" />)
    const loaderInst = loader.root.findByType(Text)

    expect(loaderInst.props.children).toBe('loading')
  })
})