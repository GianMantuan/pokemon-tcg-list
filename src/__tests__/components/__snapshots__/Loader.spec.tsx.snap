// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Component should render Loader component 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size={36}
  />
</View>
`;
